nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you also add the error message of the **NotImplementedError** so the user now what is not implemented. You can update it to something like this:
```ruby
 raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
```